src/api/index.js

```diff
@@ -0,0 +1,30 @@
+import apiClient from 'panoptes-client/lib/api-client';
+
+export const getAllUserClassifications = async (userId) => {
```

Comment:
Copying @coryzoon comment relevant to this discussion into this thread:
> As far as the user classifications those should be running asynchronously and should not block the user, however, if you have a user that has a lot of classifications that I could test with, it would help me confirm. Also I had a discussion with @lcjohnso about the classifications and I believe it was agreed to not use the ERAS endpoint at this time (Cliff, please correct me if I misunderstood this). I get the classifications anytime the project list on the home screen is refreshed, so I want to make sure I'm avoiding any potential performance issues.

Clarifying past discussions with Cory about stats: stats came up in regards to two issues related to the user ribbon (https://github.com/zooniverse/mobile/issues/265; primary issue for documentation purposes) and classification count updates (https://github.com/zooniverse/mobile/issues/216). My statement at the time was that we should prioritize other work over stat reporting improvements and a migration to ERAS.  While I was aware of Cory's use of the API to query classification counts and identify new projects where the user has participated, I didn't realize / comprehend what endpoint Cory was querying and the potential issue regarding users with large classification counts.

I agree with Mark -- the current implementation that uses the classification endpoint is not a workable solution. One alternative would be to mimic the method used by the user ribbon, querying the user project preferences (where one iterates over projects rather than individual classification), however even this has been shown to run slow for superusers.  My recommendation would be to use the new ERAS service instead of the Panoptes API's classification endpoint -- specifically the authenticated user classification endpoint (see https://github.com/zooniverse/eras/wiki/API-Callout-Examples#example-requests-2)

Example call from those docs:

    ## query to get total_count and project_contributions (and counts per project) of all classifications of user with id 1
    GET https://eras-staging.zooniverse.org/classifications/users/1?project_contributions=true

Reply:
I pushed a change that uses the ERAS endpoint above. I'm going to continue working on this to see if i can use the panoptes-client to achieve this instead of an HTTP request to the endpoint. @mcbouslog I looked at the frontend code base for a similar implementation but couldn't find anything. If you are aware of any snippets that might help me with this, please let me know.

Reply:
@coryzoon -- Check out the `eras-client` that is available within the panoptes-javascript-client: https://github.com/zooniverse/panoptes-javascript-client/blob/main/lib/eras-client.js

That library is used in the Panoptes-Front-End for project stats pages ([link to code](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/pages/project/stats/stats.jsx)), so might be helpful to you.

Reply:
@lcjohnso Looking at https://github.com/zooniverse/eras/wiki/API-Callout-Examples#example-requests-2 it seems like the example call you pasted above is exactly what I need as it gives me all of the project ids that the user has classified on. I'm not seeing that panoptes-client supports this endpoint. The example above looks like it's trying to get /classifications?workflow_id=x&period=x whereas I'm trying to get /classifications/user/x?project_contributions=true. In eras-client.js I don't see where it would pass the project_contributions=true param to the http request. Maybe @mcbouslog can confirm if this should be possible or not. I do currently have it working without using the panoptes-client so if it's not supported yet we could proceed with what I have and then update it later once it's available in panoptes-client.

Reply:
There's no need to use `panoptes-client`/`eras-client`, but wanted you to be aware of that piece of the client if only just as a reference for your custom code.

@mcbouslog might want to comment on whether he will be wanting to use `eras-client` to query `/classifications/users/X` with `project_contributions` query param -- perhaps it is worth it to add that funcationality now to the `panoptes-client` for use in mobile and later for FEM user stats pages.  But from my perspective, it is also OK if Cory writes one-off code for use in mobile that doesn't contribute to `eras-client` -- we don't necessarily need to make all client-like code reusable.

Reply:
`panoptes-client/eras-client` doesn't support the noted `project_contributions` param or credentialed requests yet (`/users/:id` or `/groups/:id`), which I think are needed for this `mobile` purpose. That functionality will be added eventually, but I agree these one-off changes are fine for now and we can refactor when the `panoptes-client/eras-client` works for this purpose.

I'm working on the `[panoptes-js](https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-panoptes-js)/erasAPI` at the moment, but that's the new client used predominantly by FEM, that's not a dependency in `mobile` at the moment.

So both of our commonly used clients, `panoptes-client` and `panoptes-js`, will soon have more ERAS functionality, but I think these one-off changes are good for now.

Reply:
[`panoptes-client`](https://www.npmjs.com/package/panoptes-client) is an old, stateful client with some odd bugs, because it's stateful. It keeps an undocumented cache of all downloaded resources, and prefers that cache to cut down on network requests. `import apiClient from panoptes-client/lib/api-client` imports an instance of the client with any previously requested resources stored in `apiClient._types` (I think, it's been a while since I looked at it.) This can lead to weird bugs where the client returns outdated copies of resources, instead of a fresh copy from the Panoptes API. There are undocumented  workarounds to bypass the internal cache, but it can be a pain if you're working with resources that change frequently (like users and user stats.)

[`@zooniverse/panoptes-js`](https://www.npmjs.com/package/@zooniverse/panoptes-js) is a newer client (about 5 years old), which is stateless, by design. It is responsible for requesting resources from Panoptes, but resource management is left to local state management in the parent app/component.

Both clients are old enough that they predate the `fetch` API in browsers and Node, and both use XHR, via superagent. With `fetch` now supported in all modern browsers, you're probably better off writing your own API code using `fetch`.